homeassistant/components/nextcloud/entity.py

```diff
@@ -21,3 +23,14 @@ def __init__(self, coordinator: NextcloudDataUpdateCoordinator, item: str) -> No
     def unique_id(self) -> str:
         """Return the unique ID for this sensor."""
         return f"{self.coordinator.url}#{self.item}"
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device specific attributes."""
+        return DeviceInfo(
+            name=self.coordinator.url,
```

Comment:
An URL doesn't seems like a user-friendly name for a device?

Reply:
Maybe config-flow should be adjusted to parse the URL, extract the hostname, and set the entry title to the hostname.

Then you can maybe use `config_entry.title` as device name.

Reply:
Or even simply `Nextcloud` as the default device name. That's more user friendly and most users will only have one anyway.